examples/experimental/dagster-airlift/dagster_airlift/core/top_level_dag_def_api.py

```diff
@@ -106,6 +106,48 @@ def asset_one() -> None: ...
     return assets_list
 
 
+def assets_with_dag_mappings(
+    dag_mappings: Mapping[str, Iterable[Union[AssetsDefinition, AssetSpec]]],
+) -> Sequence[Union[AssetsDefinition, AssetSpec]]:
+    """Modify assets to be associated with a particular dag in Airlift tooling.
+
+    Used in concert with `build_defs_from_airflow_instance` to observe an airflow
+    instance to monitor the dags that are associated with the assets and
+    keep their materialization histories up to date.
```

Comment:
I would add something liike:

"In contrast with assets_with_task_mappings, which is on a per-task basis, this is used in concert with (API on in_airflow side) dag-level mappings where an entire DAG is migrated at once."